### ISSUE TITLE
Completed task checkbox drops collection at commit  f11a875b

### ISSUE THREAD

**Issue by jeremybrenner** [meteor/simple-todos-angular#2]
<strong>Update:</strong> After a little debugging, it seems this error was being caused by the incompleteCount function. Specifically, calling the query: <code>return Tasks.find({ checked: {$ne: true} });</code> with <code>incompleteCount()</code> in the view. 

A solution that fixes this bug is:

in <strong>simple-todos-angular.js</strong> bind the filtered meteor collection to a <code>$scope</code> variable with <code>$meteor.collection</code>

```
    $scope.incompleteCount = $meteor.collection(function() {
        return Tasks.find({ checked: {$ne: true} });
      });
```

then in <strong>todos-list.ng.html</strong>, render the length of the collection in the context of the variable

```
<h1>Todo List ( {{ incompleteCount.length }} )</h1>
```

This allows this value to react dynamically on the page, without creating a conflict!

---

I wasn't able to debug the cause entirely (working on that now), but it seems there is some sort of $scope conflict that occurs when you toggle the "Hide Completed Tasks" box on and off, then change the state of the boxes for an individual task( or create a new task), and try to hide again. The $scope.query is cycling just fine, but something is causing the hide box to subsequently drop the whole collection. This seems to resolve once the validation and CRUD methods are are implemented instead of the helpers( and after watching the console it seems to be related to class changes), but just wanted to point it out! To replicate this bug:
1. Clone and run app at commit "Add an option to hide completed tasks, and show incomplete task count" ( f11a875b19ccf321e62fd5494d8de5d03df0c55e )
2. Enter three tasks, check one as finished.
3. Check the hide box ( checked task will disappear )
4. Uncheck the hide box ( hidden task returns properly )
5. Uncheck the task in the list. 
6. Toggle hide box again.
7. Whole collection is removed from page and dropped from db (verified in Mongo shell)

Screenshots of bug in action:

![screen shot 2015-07-03 at 7 29 33 pm](https://cloud.githubusercontent.com/assets/11710882/8506132/fe12e10a-21b9-11e5-8e43-33eec5022f98.png)
![screen shot 2015-07-03 at 7 29 41 pm](https://cloud.githubusercontent.com/assets/11710882/8506133/0416c378-21ba-11e5-80a5-d2aee23f8ad9.png)
![screen shot 2015-07-03 at 7 29 49 pm](https://cloud.githubusercontent.com/assets/11710882/8506134/07fd07ea-21ba-11e5-89d7-d6818ad8114b.png)
![screen shot 2015-07-03 at 7 30 04 pm](https://cloud.githubusercontent.com/assets/11710882/8506135/0bf7038c-21ba-11e5-9571-cb0e4fd53f9e.png)
![screen shot 2015-07-03 at 7 30 12 pm](https://cloud.githubusercontent.com/assets/11710882/8506136/0e5f4602-21ba-11e5-887e-ab86ba3b8e12.png)

**Comment by jeremybrenner** [comment]
Update: This is being caused by the collection query in the 'incompleteCount' function!

**Comment by Urigo** [comment]
That's a very interesting case.
And I really like your solution, very angular way, but I want to understand first why this happens with the old incomplete function.
Looking on it now

**Comment by Urigo** [comment]
@jeremybrenner looks like this issue was fixed in the `urigo:angular` package itself.

Can you try running everything as before just with version `0.9.2-alpha.3` ?

in your packages file, add `@0.9.2-alpha.3` to `urigo:angular` so it will look like `urigo:angular@0.9.2-alpha.3`

Let me know it that helped and I'll release the new version right away

**Comment by Urigo** [comment]
That was interesting...
So, it's true that `incompleteCount` caused that error, but it shouldn't have.
It made Angular's digest cycle run more times, but we shouldn't care how frequently it is called.

The problem was that `unregisterAutoBind` function which stops Angular `$watch` is not synchronous and another `$watch` call was being called after we called the `stop()` function on it. 

So I've made sure that won't happen with a flag and I keep looking if there is anyway to make sure the `$watch` has actually stopped (I'm not sure there is a way right now, maybe an Angular pull request...)

**Comment by Urigo** [comment]
@jeremybrenner please try running this version and see if that works for you:

```
urigo:angular@0.9.2-alpha.4
```

**Comment by jeremybrenner** [comment]
@Urigo Thanks for the explanation- I spent a lot of time looking at this and was very curious about what was going on. Using the 0.9.2-alpha.4 version fixes the bug, awesome work!

**Comment by Urigo** [comment]
Great, I've just released 0.9.2 that includes that fix
